project-base/app/src/DataFixtures/Demo/DemoDataFactory/ProductDemoDataFactory.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\DataFixtures\Demo\DemoDataFactory;
+
+use App\DataFixtures\Demo\UnitDataFixture;
+use App\DataFixtures\Demo\VatDataFixture;
+use App\Model\Product\ProductData;
+use App\Model\Product\ProductDataFactory;
+use App\Model\Product\Unit\Unit;
+use DateTime;
+use Shopsys\FrameworkBundle\Component\DataFixture\PersistentReferenceFacade;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Pricing\Vat\Vat;
+
+class ProductDemoDataFactory
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \App\Model\Product\ProductDataFactory $productDataFactory
+     * @param \Shopsys\FrameworkBundle\Component\DataFixture\PersistentReferenceFacade $persistentReferenceFacade
+     */
+    public function __construct(
+        private readonly Domain $domain,
+        private readonly ProductDataFactory $productDataFactory,
+        private readonly PersistentReferenceFacade $persistentReferenceFacade,
+    ) {
+    }
+
+    /**
+     * @return \App\Model\Product\ProductData
+     */
+    public function createDefaultData(): ProductData
+    {
+        $productData = $this->productDataFactory->create();
+
+        $this->setVat($productData, VatDataFixture::VAT_HIGH);
+        $this->setUnit($productData, UnitDataFixture::UNIT_PIECES);
+        $this->setSellingTo($productData, null);
+
```

Comment:
I feel like there could be more data set in this method, I would try to spot the fields that are set multiple times with the same value, and use that value as the default one. Then, only products with different values would set their values appropriately. I see that means that some values will be set two times (in `createDefaultData` and then again with the unique value) but I believe it would not slow down the data fixtures loading (or at least not significantly).

To give an example - `sellingFrom` is set for the same date at least ten times so we could get rid of all those lines if the value is default:

![Screenshot from 2024-05-31 20-21-50](https://github.com/shopsys/shopsys/assets/10401898/a043ae87-5757-4ca2-83e4-703a5828af5f)

Reply:
Removed more lines where unnecessary empty value / array setting was happening, but i don't think there is much more we can get out of this approach, remaining data is just too varying in general.

Regarding sellingFrom i did not implemented that yet. There is at most 29 same values and trade-off between trimming down 29 lines and overriding value at 120+ other occurrences does not seem worth it to me, but i can do it if you think otherwise.

Reply:
ok, just move the `setSellingFrom` to `ProductDemoDataSetter` (with non-nullable `$date` parameter)

Reply:
done, changed setSellingTo to non-nullable as well